Cleanup available settings, stage for proper options handling

jarolrod opened the pull request:
based on bitcoin-core/gui-qml#156 and bitcoin-core/gui-qml#157
This primarily cleans up settings/options that cannot be modified by the OptionsModel we will use; thus staging the gui for proper options handling. 

Introduces a static backup for the storage lottie element. The SVG was provided by @GBKS, and the png generated with inkscape. The resulting png was then run through the optimize python script, the output of the script is as follows:
```
optimizing storage-dark.png... done
optimizing storage-light.png... done
summary:
+++++++++++++++++
storage-dark.png
  size diff from: 10942 to: 8412
  old sha256: ff0937aabffa949a241df90ff3a61d9e3cb35a836a11e497718196097b0e68fb
  new sha256: 8a3c195ee3b44624c89ab648020005a0e955b7e6086a89c966de0547f337b98b

storage-light.png
  size diff from: 11205 to: 8693
  old sha256: f7e707cd997f8acf3ac273f6680611f420375ed40cdbacb0a18e2003d49496d4
  new sha256: 38ad2e515a55980d01cfeaeee0bb7264005a739e65fc57f4bb99cdb92e39f3c6

completed. Checksum stable: False. Total reduction: 5042 bytes
```

# Master
| Developer Options | Storage Location | Storage | 
| ------------------------- | ----------------------- | ----------- | 
| ![master-2](https://user-images.githubusercontent.com/23396902/184556306-e23b4a7a-2d7a-44ea-847c-481fe559ecf8.png) | ![master-3](https://user-images.githubusercontent.com/23396902/184556309-4519c980-6617-4809-b66e-f232c800e58d.png) | ![master-4](https://user-images.githubusercontent.com/23396902/184556344-6f5eceb2-51d1-4778-b359-e60ea14f339b.png) | 

| Storage Settings | Connection | Connection Settings |
| --------------------- | --------------- | --------------------------- |
| ![master-5](https://user-images.githubusercontent.com/23396902/184556353-c34b5cba-f3b3-4cf9-a2ff-e1178ccdf0dc.png) | ![master-6](https://user-images.githubusercontent.com/23396902/184556356-806f1d7d-1a98-450e-9b1b-f33442bf0340.png) | ![master-7](https://user-images.githubusercontent.com/23396902/184556359-49162b4b-6eda-413d-9cdf-865a724fb865.png) |

# PR
| Developer Options | Storage Location | Storage | 
| ------------------------- | ----------------------- | ----------- | 
| ![newdev-screen](https://user-images.githubusercontent.com/23396902/185256655-f718c4b1-26a4-487b-9eff-9d8937d42db2.png) | ![pr-3](https://user-images.githubusercontent.com/23396902/184556393-a8abe643-7aea-4c3a-82bf-24391c48a32e.png) | ![pr-4](https://user-images.githubusercontent.com/23396902/184556401-39117b85-a98d-4a7a-8d51-6013d97e13b7.png) |


| Storage Settings | Connection | Connection Settings |
| --------------------- | --------------- | --------------------------- |
| ![pr-5](https://user-images.githubusercontent.com/23396902/184556409-ab430f04-387e-4f22-b722-ff681a71fe8c.png) | ![new-ibd-screen](https://user-images.githubusercontent.com/23396902/185256895-1f356346-75ed-4f2c-b7a3-a094d50f9987.png) | ![newcon-screen](https://user-images.githubusercontent.com/23396902/185256796-d12f7cc3-4824-41d1-9c4b-e593d2d2a707.png) |


[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/158)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/158)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/158)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/158)

jarolrod commented:
updated from 674117d to be2c533

changes: rebased over changes on master, dropped unneeded PR dependencies

jarolrod commented:
converted to draft until we are able to have a discussion on the bitcoin core app call with designers

jarolrod commented:
updated from be2c533 to 16fa2d4

Changes: based on new PR's to not conflict, used correct storage image

GBKS commented:
Awesome progress.

jarolrod commented on src/qml/pages/onboarding/onboarding06a.qml:
```diff
@@ -24,22 +24,27 @@ Page {
             Layout.alignment: Qt.AlignCenter
             Layout.topMargin: 30
-            text: "Detailed Settings"
+            text: "Connection Settings"
```
addressed in 0f969b8 , thanks!

jarolrod commented:
updated from 16fa2d4 to 0f969b8 

changes: rebased over main, moved `dbcache` and `scripthreads` settings from Connection settings to Developer settings, corrected a [capitalization error](https://github.com/bitcoin-core/gui-qml/pull/158#discussion_r947993292)

shaavan commented on src/qml/components/DeveloperOptions.qml:
```diff
@@ -19,14 +19,16 @@ ColumnLayout {
     }
     Setting {
         Layout.fillWidth: true
-        header: qsTr("Network")
```
4. I was able to verify the non-modifiability by the optionsModel of all the removed settings except this one (”Network”), so could you provide some details on the file responsible for handling the chain switching?

jarolrod replied:
there is no file responsible for handling chain switching, that isn't something we can do as of now

shaavan commented on src/qml/components/StorageSettings.qml:
```diff
@@ -27,13 +27,5 @@ ColumnLayout {
         actionItem: ValueInput {
             description: qsTr("c://.../data")
         }
```
5. The storageLocation.qml component allows selecting the location for data storage. So providing the same option in the StorageSettings during the onboarding process seems redundant.

jarolrod replied:
Yes as it should be, the first screen you reference is the easy path settings, the next is a detailed settings screen.

shaavan replied:
@jarolrod

Thanks for providing the clarification. But I still have some points to discuss.

The **storage location** page provides two options: The **Default Directory** and the **Custom Directory**. And as per my understanding, clicking on the **Custom Directory** will open the file manager to allow the user to choose the directories location. So having the exact same option in the storage setting still seems redundant to me.

Also, the Storage Settings page deals with the size-related factors regarding storage, so having a storage location option seems out of place when we have a dedicated onboarding page for the storage location selection.

I think it is better to keep the functionality of each page mutually exclusive and avoid overlap in their usage.

These are my thoughts on this matter, but I agree with your reasoning [here](https://github.com/bitcoin-core/gui-qml/pull/158#issuecomment-1223512830), and removing UI-based redundancies is out of the scope of this PR, so we can further brainstorm on this with the designer as well.

shaavan reviewed (commented):
Concept ACK

Fantastic work, @jarolrod! 

I have some points to discuss/add that can further improve this clean-up.

1. The ConnectionOptions.qml file is now irrelevant and can be removed.

2. In StorageSettings.qml:

```qml
Setting {
        Layout.fillWidth: true
        header: qsTr("Store Recent blocks only")
        actionItem: OptionSwitch {}
    }
    Setting {
        Layout.fillWidth: true
        header: qsTr("Storage limit")
        actionItem: ValueInput {
            description: qsTr("2 GB")
        }
    }
```
Doesn’t giving a "storage limit" imply that only the recent block will be stored? So is providing a separate option for "store only recent blocks" correct?

jarolrod commented on src/qml/components/ConnectionSettings.qml:
```diff
@@ -11,42 +11,24 @@ ColumnLayout {
-        description: qsTr("Increases data usage")
+        header: qsTr("Map port using UPnP")
         actionItem: OptionSwitch {}
     }
     Setting {
         Layout.fillWidth: true
-        header: qsTr("Blocks Only")
-        description: qsTr("Reduces data usage.")
+        header: qsTr("Map port using NAT-PMP")
```
yes, they can be enabled at the same time

jarolrod commented:
@shaavan 
> The ConnectionOptions.qml file is now irrelevant and can be removed.

Yes, but that isn't going to be removed until there's absolutely no need for it in the future. This is useful when porting to mobile

> Doesn’t giving a "storage limit" imply that only the recent block will be stored? So is providing a separate option for "store only recent blocks" correct?

This PR doesn't intend to solve all issues around the workings of the settings themselves. Just cleaning up from fake settings to the real ones. This is correct as is. The UI issue here is that the value input field for the prune size should be disabled until the prune button is enabled. The UI pattern to use there is being discussed with designers.

shaavan commented:
> Yes, but that isn't going to be removed until there's absolutely no need for it in the future. This is useful when porting to mobile

That seems reasonable.

> The UI issue is that the value input field for the prune size should be disabled until the prune button is enabled.

Maybe this can be solved by adding an "is_enable" property to the setting control, which displays the content of settings but makes them unclickable. But I have to further experiment with this idea to see if it will work.

jarolrod commented:
> Maybe this can be solved by adding an "is_enable" property to the setting control, which displays the content of settings but makes them unclickable. But I have to further experiment with this idea to see if it will work.

Yeah it's straight forward to do this with binding, just something to figure out on the next iterative step, when there is actually options handling going on

jarolrod commented:
updated from[ 0f969b8 ](https://github.com/bitcoin-core/gui-qml/commit/0f969b8df5bf02e150198210918ba2b48b772aab)to[ d81c43d](https://github.com/bitcoin-core/gui-qml/commit/d81c43dd6f1fa901760f8d9d6914722522d5aa8a)

changes: rebased over master and fixed conflicts

hebasto reviewed (approved):
ACK d81c43dd6f1fa901760f8d9d6914722522d5aa8a, I have reviewed the code and it looks OK, I agree it can be merged.

hebasto commented on src/qml/components/StorageOptions.qml:
```diff
@@ -15,7 +15,7 @@ ColumnLayout {
         ButtonGroup.group: group
         text: qsTr("Reduce storage")
-        description: qsTr("Uses about 75GB.")
+        description: qsTr("Uses about 2GB.")
```
In other places we use a space to separate a unit from a value.
